playbooks/healthchecker.yaml

```diff
@@ -0,0 +1,59 @@
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations
+# under the License.
+---
+- name: Install healthchecker
+  hosts: healthchecker
+  become: yes
+  tasks:
+    - name: New a healthchecker directory
+      file:
+        path: /etc/healthchecker/
+        mode: 0755
+        state: directory
+
+    - name: Copy necessary scripts
+      template: src={{ item.src }} dest={{ item.dest }} mode=0755
+      with_items:
+        - { src: "{{ labkeeper_config_git_dest }}/healthchecker/healthchecker.sh", dest: '/etc/healthchecker/healthchecker.sh'}
+        - { src: "{{ labkeeper_config_git_dest }}/healthchecker/refresh.py", dest: '/etc/healthchecker/refresh.py' }
+        - { src: "{{ labkeeper_config_git_dest }}/healthchecker/fix.py", dest: '/etc/healthchecker/fix.py'}
+        - { src: "{{ labkeeper_config_git_dest }}/healthchecker/switch.py", dest: '/etc/healthchecker/switch.py' }
+      vars:
+        zk_cli_conf: "/etc/openlab/openlab.conf"
+        heartbeat_internal: 100
+        ori_master_ip: "{{ hostvars[groups['zuul-web-master'][0]].ansible_host }}"
+        ori_backup_ip: "{{ hostvars[groups['zuul-web-slave'][0]].ansible_host }}"
+        slave_ip: "{{ hostvars[groups['zuul-web-slave'][0]].ansible_host }}"
+        slave_backup_ip: "{{ hostvars[groups['zuul-web-slave'][0]].ansible_host }}"
+
+    - name: install script dependency
+      pip:
+        name: ['six', 'iso8601', 'requests', 'configparser', 'PyGithub', 'openlabcmd']
+
+    - name: config healthchecker
+      copy: src={{ item.src }} dest={{ item.dest }}
+      with_items:
+        - { src: "{{ labkeeper_config_git_dest }}/healthchecker/healthchecker.timer", dest: '/etc/systemd/system/healthchecker.timer'}
+        - { src: "{{ labkeeper_config_git_dest }}/healthchecker/healthchecker.service", dest: '/etc/systemd/system/healthchecker.service' }
+
+    # As systemd ansible module bugs not fix #36585 and #48786
```

Comment:
Really? Have you tested it? I see 36585 is closed https://github.com/ansible/ansible/issues/36585 and 48786 has some fixed comment.

Reply:
Cool. good to know